write-fonts/src/tables/glyf.rs

```diff
@@ -487,4 +689,39 @@ mod tests {
             }
         )
     }
+
+    #[test]
+    fn roundtrip_composite() {
+        let font = FontRef::new(test_data::test_fonts::VAZIRMATN_VAR).unwrap();
+        let loca = font.loca(None).unwrap();
+        let glyf = font.glyf().unwrap();
+        let read_glyf::Glyph::Composite(orig) = loca.get_glyf(GlyphId::new(2), &glyf).unwrap().unwrap() else { panic!("not a composite glyph") };
+
+        let mut iter = orig
+            .components()
+            .map(|comp| Component::new(comp.glyph, comp.anchor, comp.transform, comp.flags));
+        let mut composite = CompositeGlyph::new(iter.next().unwrap());
+        composite.add_component(iter.next().unwrap());
+        composite._instructions = orig.instructions().unwrap_or_default().to_vec();
+        //FIXME: figure out how we're actually computing bboxes here
+        composite.bbox = Bbox {
+            x_min: orig.x_min(),
+            y_min: orig.y_min(),
+            x_max: orig.x_max(),
+            y_max: orig.y_max(),
+        };
+        assert!(iter.next().is_none());
+        let bytes = crate::dump_table(&composite).unwrap();
+        let ours = read::tables::glyf::CompositeGlyph::read(FontData::new(&bytes)).unwrap();
+
+        let our_comps = ours.components().collect::<Vec<_>>();
+        let orig_comps = orig.components().collect::<Vec<_>>();
+        assert_eq!(our_comps.len(), orig_comps.len());
+        assert_eq!(&our_comps[0], &orig_comps[0]);
+        assert_eq!(&our_comps[1], &orig_comps[1]);
```

Comment:
I'm failing to spot where we assured ourselves there would be 2; perhaps we should additionally assert len is 2?